tests/integration/framework/process/daprd/daprd.go

```diff
@@ -327,17 +328,41 @@ func (d *Daprd) ProfilePort() int {
 	return d.profilePort
 }
 
+type MetricBucket struct {
+	UpperBound      float64
+	CumulativeCount uint64
+}
+
+type metricsOptions struct {
+	includeBuckets bool
+}
+
+func WithIncludeBuckets() func(metricsOptions *metricsOptions) {
+	return func(m *metricsOptions) {
+		m.includeBuckets = true
+	}
+}
+
 // Returns a subset of metrics scraped from the metrics endpoint
-func (d *Daprd) Metrics(t *testing.T, ctx context.Context) map[string]float64 {
+func (d *Daprd) Metrics(t *testing.T, ctx context.Context, opts ...func(*metricsOptions)) map[string]float64 {
 	t.Helper()
 
+	curOpts := metricsOptions{}
+	for _, o := range opts {
+		o(&curOpts)
+	}
+
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("http://%s/metrics", d.MetricsAddress()), nil)
 	require.NoError(t, err)
 
 	resp, err := d.httpClient.Do(req)
 	require.NoError(t, err)
 	require.Equal(t, http.StatusOK, resp.StatusCode)
 
+	// read resp.Body to variable
+	body, _ := io.ReadAll(resp.Body)
+	resp.Body = io.NopCloser(bytes.NewReader(body))
```

Comment:
Why are we doing this? Seems superfluous :raised_eyebrow:

Reply:
yeah, that was me debugging, let me remove